src/main/webapp/assets/js/view/application-add-wizard.js

```diff
@@ -680,8 +680,13 @@ define([
                 this.$('.required-config-loading').hide()
                 if (catalogEntryItem!=null && catalogEntryItem.config!=null) {
                     var that = this
-                    _.each(catalogEntryItem.config, function (cfg) {
-                        if (cfg.priority !== undefined) {
+                    _.chain(catalogEntryItem.config).sortBy(function (cfg){
+                        return cfg.priority ? cfg.priority : 99999;
+                    }).each(function (cfg) {
+                        if (cfg.pinned === true) {
+                            var html = _.template(RequiredConfigEntryHtml, {data:cfg});
+                            that.$('.config-table').append(html);
+                        } else if (cfg.constraints.indexOf('required') > - 1 && (!cfg.defaultValue || cfg.defaultValue === '')) {
```

Comment:
> '' is treated as false

Are you sure of that? I always check if a variable exists **and** different than empty because I had troubles in the past. I would update for:
```js
if (cfg.pinned === true || (typeof cfg.defaultValue === 'string' && cfg.defaultValue !== '')) {
    var html = _.template(RequiredConfigEntryHtml, {data:cfg});
    that.$('.config-table').append(html);
}
```